src/mitol/payment_gateway/api.py

```diff
@@ -361,6 +433,106 @@ def prepare_checkout(
             "method": "POST",
         }
 
+    def get_client_configuration(self):
+        """
+        Get the configuration required for CyberSource API client
+        """
+        configuration_dictionary = {
+            "authentication_type": "http_signature",
+            "merchantid": settings.MITOL_PAYMENT_GATEWAY_CYBERSOURCE_MERCHANT_ID,
+            "run_environment": CYBERSOURCE_LIVE_API_URL
+            if settings.ENVIRONMENT == "prod"
+            else CYBERSOURCE_TEST_API_URL,
+            "merchant_keyid": settings.MITOL_PAYMENT_GATEWAY_CYBERSOURCE_MERCHANT_SECRET_KEY_ID,
+            "merchant_secretkey": settings.MITOL_PAYMENT_GATEWAY_CYBERSOURCE_MERCHANT_SECRET,
+            "timeout": 1000,
+        }
+        return configuration_dictionary
+
+    def perform_refund(self, refund):
+        """
+        Based on the provided refund detail data object, this method calls the cyber source refund API call after
+        performing the appropriate operation on the data.
+
+        returns:
+
+            i) API Success:
+                Returns a ProcessorResponse object that the caller app can use to perform whatever they need
+
+            ii)  API Failure ("DUPLICATE_REQUEST"):
+                Raises RefundDuplicateException when the refund API call is duplicate
+
+            iii)  API Failure (General):
+                Raises the exception(Mostly CyberSource.rest.ApiException) to the caller to decide what their
+                application behaviour
+        """
+
+        api_instance = RefundApi(self.get_client_configuration())
+        refund_payload = self.generate_refund_payload(refund)
+        transaction_id = refund.transaction_id
+
+        try:
+            return_data, status, body = api_instance.refund_payment(
+                refund_payload, transaction_id
+            )
+            _body = json.loads(body)
+            # Transforming the response in a format that process response decoder can work on while keeping our
+            # structure consistent
+            _response_transformed = {
```

Comment:
I'd try to rename these to avoid `_` prefixed variables, just because those usually denote private variables and all variables are private to a function so it's a bit odd.

Reply:
Addressed